pilot/pkg/controller/workloadentry/workloadentry_controller_test.go

```diff
@@ -137,17 +137,27 @@ func TestAutoregistrationLifecycle(t *testing.T) {
 		})
 		t.Run("different instance", func(t *testing.T) {
 			// disconnect, make sure entry is still there with disconnect metadata
-			c1.QueueUnregisterWorkload(p)
+			c1.QueueUnregisterWorkload(p, time.Now())
 			time.Sleep(features.WorkloadEntryCleanupGracePeriod / 2)
 			checkEntryOrFail(t, store, wgA, p, "")
 			// reconnect, ensure entry is there with the new instance id
 			c2.RegisterWorkload(p, time.Now())
 			checkEntryOrFail(t, store, wgA, p, c2.instanceID)
 		})
+		t.Run("slow processed disconnect", func(t *testing.T) {
+			// reconnect, ensure entry is there with the new instance id
+			reconnectAt := time.Now()
+			c2.RegisterWorkload(p, reconnectAt)
+			checkEntryOrFail(t, store, wgA, p, c2.instanceID)
+			// disconnect for orignal connection processed late, make sure entry is unmodified
+			c1.QueueUnregisterWorkload(p, reconnectAt.Add(-1*time.Second))
```

Comment:
outside of scope for this PR, but the agent has a (default) 5s termination period. In this 5s, we can easily send a "I am shutting down" signal to istiod, which can improve our unregister workflow?

We still need everything we have today though, as there can be hard terminations, but perhaps the common case is improved?